src/lib/rnp.cpp

```diff
@@ -6173,6 +6173,64 @@ try {
 }
 FFI_GUARD
 
+rnp_result_t
+rnp_key_get_default_key(rnp_key_handle_t  primary_key,
+                        const char *      usage,
+                        uint32_t          flags,
+                        rnp_key_handle_t *default_key)
+try {
+    if (!primary_key || !usage || !default_key) {
+        return RNP_ERROR_NULL_POINTER;
+    }
+    uint8_t keyflag = 0;
+    bool    no_primary = false;
+    if (!str_to_key_flag(usage, &keyflag)) {
+        return RNP_ERROR_BAD_PARAMETERS;
+    }
+    if (flags & RNP_KEY_SUBKEYS_ONLY) {
+        no_primary = true;
+        flags &= ~RNP_KEY_SUBKEYS_ONLY;
+    }
+    if (flags) {
+        FFI_LOG(primary_key->ffi, "Invalid flags: %" PRIu32, flags);
+        return RNP_ERROR_BAD_PARAMETERS;
+    }
+    pgp_key_t *key = get_key_prefer_public(primary_key);
+    if (!key) {
+        return RNP_ERROR_BAD_PARAMETERS;
+    }
+    pgp_key_t *defkey = find_suitable_key(
+      PGP_OP_UNKNOWN, key, &primary_key->ffi->key_provider, keyflag, no_primary);
+    if (!defkey) {
+        *default_key = NULL;
+        return RNP_ERROR_NO_SUITABLE_KEY;
+    }
+    pgp_key_search_t search = {(pgp_key_search_type_t) 0};
```

Comment:
Why not to reuse `rnp_locate_key()` here?

Reply:
Because it requires excess conversions binary->hex->binary, see https://github.com/rnpgp/rnp/issues/1556

Reply:
Posted a PR #1560 which should fix that, and solution may be reused in this PR.

Reply:
Still not using rnp_locate_key_int(), as it allocates memory first, and presence of private key can be checked later on return, so in case private key is required by specified key usage but it is not present, memory must be free-d.

Reply:
@antonsviridenko This could be easily addressed by adding parameter like `require_secret` to the `rnp_locate_key_int()`. Anyway, let's merge this and address this later. Thanks!